src/main/java/io/debezium/connector/yugabytedb/YugabyteDBErrorHandler.java

```diff
@@ -30,6 +33,13 @@ protected boolean isRetriable(Throwable throwable) {
             return true;
         }
 
+        // This condition will be hit when the poller thread indicates that a new table has been
+        // found to be associated with the provided stream ID.
+        if (throwable instanceof NewTableFoundException) {
```

Comment:
Don't understand the workflow. Thread throws an exception and this part gets called? Is that why you use thread instead of executor service?